### PR TITLE
Introduce LibraryMembership Index

### PR DESCRIPTION
**Pull request by leogrim**
@eishay @andrewconner I may be able to pull this off tonight
Once this index is built, I'll switch search over and we can stop reindexing all library memberships with every new one.

### CODE REVIEW

**Comment by eishay** [comment]
great!

**Comment by yingjieMiao** [comment]
I assume we need to change the index query? test?

**Comment by yingjieMiao** [comment]
I am not clear about the search performance impact of index structure change. let's keep that in mind.

**Comment by leogrim** [comment]
No you'll see my next pull request later tonight, pretty minimal change as far as queries are concerned :)

**Comment by andrewconner** [comment]
Amazing job, that was fast

File: kifi-backend/modules/search/app/com/keepit/search/index/graph/library/membership/LibraryMembershipIndexable.scala [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    doc.add(buildKeywordField(userField, membership.userId.id.toString))
+    doc.add(buildIdValueField(userIdField, membership.userId))
+    if (membership.showInSearch) {
+      doc.add(buildKeywordField(searcherField, membership.userId.id.toString))
+    }
+
+    doc.add(buildKeywordField(libraryField, membership.libraryId.id.toString))
+    doc.add(buildIdValueField(libraryIdField, membership.libraryId))
+
```
Comment (yingjieMiao) [inline kifi-backend/modules/search/app/com/keepit/search/index/graph/library/membership/LibraryMembershipIndexable.scala:54]: what are the differences of these two fields? are they going to be queried differently?

Comment (leogrim) [inline kifi-backend/modules/search/app/com/keepit/search/index/graph/library/membership/LibraryMembershipIndexable.scala:54]: One is indexed, the other one is stored.

**Pull request merged**